src/cargo/util/command_prelude.rs

```diff
@@ -986,6 +1001,44 @@ pub fn root_manifest(manifest_path: Option<&Path>, gctx: &GlobalContext) -> Carg
     }
 }
 
+pub fn lockfile_path(
+    lockfile_path: Option<&Path>,
+    gctx: &GlobalContext,
+) -> CargoResult<Option<PathBuf>> {
+    let Some(lockfile_path) = lockfile_path else {
+        return Ok(None);
+    };
+
+    gctx.cli_unstable()
+        .fail_if_stable_opt("--lockfile-path", 5707)?;
+
+    let path = gctx.cwd().join(lockfile_path);
+
+    if !path.ends_with(LOCKFILE_NAME) {
+        bail!("the lockfile-path must be a path to a {LOCKFILE_NAME} file")
+    }
+    if path.is_dir() {
+        bail!(
+            "lockfile path `{}` is a directory but expected a file",
+            lockfile_path.display()
+        )
+    }
+    if !path.exists() {
+        // Root case should already be covered above
+        let parent_path = lockfile_path.parent().expect("lockfile path can't be root");
+
+        // Fail early if parent path existence can't be verified (e.g. due to permissions)
+        parent_path.try_exists().with_context(|| {
+            format!(
+                "failed to fetch lock file's parent path metadata {}",
```

Comment:
If we keep this error description, we should frame it in terms of the user (which the comment focuses on) rather than the implementation

Reply:
[Removed the check](https://github.com/rust-lang/cargo/pull/14326#discussion_r1717333464)